sdk/identity/identity/test/internal/node/clientAssertionCredential.spec.ts

```diff
@@ -0,0 +1,139 @@
+      ConfidentialClientApplication.prototype,
+      "acquireTokenByClientCredential"
+    );
+  });
+  afterEach(async function () {
+    await cleanup();
+    Sinon.restore();
+  });
+
+  it("Should throw if the parameteres are not correctly specified", async function () {
+    const errors: Error[] = [];
+    try {
+      new ClientAssertionCredential(
+        undefined as any,
+        env.AZURE_CLIENT_ID ?? "client",
+        async () => "assertion"
+      );
+    } catch (e: any) {
+      errors.push(e);
+    }
+    try {
+      new ClientAssertionCredential(
+        env.AZURE_TENANT_ID ?? "tenant",
+        undefined as any,
+        async () => "assertion"
+      );
+    } catch (e: any) {
+      errors.push(e);
+    }
+    try {
+      new ClientAssertionCredential(
+        env.AZURE_TENANT_ID ?? "tenant",
+        env.AZURE_CLIENT_ID ?? "client",
+        undefined as any
+      );
+    } catch (e: any) {
+      errors.push(e);
+    }
+    try {
+      new ClientAssertionCredential(undefined as any, undefined as any, undefined as any);
+    } catch (e: any) {
+      errors.push(e);
+    }
+    assert.equal(errors.length, 4);
+    errors.forEach((e) => {
+      assert.equal(
+        e.message,
+        "ClientAssertionCredential: tenantId, clientId, and clientAssertion are required parameters."
+      );
+    });
+  });
+
+  it("Sends the expected parameters", async function () {
+    const tenantId = env.AZURE_TENANT_ID ?? "tenant";
+    const clientId = env.AZURE_CLIENT_ID ?? "client";
+    const certificatePath = env.AZURE_CLIENT_CERTIFICATE_PATH ?? "certificate-path";
+    const authorityHost = `https://login.microsoftonline.com/${tenantId}`;
+
+    async function getAssertion(): Promise<string> {
+      const jwtoken = await createJWTTokenFromCertificate(authorityHost, clientId, certificatePath);
+      return jwtoken;
+    }
+    const credential = new ClientAssertionCredential(tenantId, clientId, getAssertion);
+
+    try {
+      await credential.getToken("https://vault.azure.net/.default");
+    } catch (e: any) {
+      // We're ignoring errors since our main goal here is to ensure that we send the correct parameters to MSAL.
+      console.log("error", e);
+    }
+
+    assert.equal(getTokenSilentSpy.callCount, 1);
+    assert.equal(doGetTokenSpy.callCount, 1);
+
+    // TODO: you can test if this matches
+    // const returnedAssertion = await getAssertion();
+    // const sentConfiguration = doGetTokenSpy.args[0][0];
+    // assert.equal(sentConfiguration.clientAssertion, "assertion");
+  });
+});
+
+
+async function createJWTTokenFromCertificate(authorityHost: string,
+  clientId: string,
+  certificatePath: string) {
+
+  const privateKeyPemCert = fs.readFileSync(certificatePath);
+  const audience = `${authorityHost}/v2.0`;
+  const secureContext = tls.createSecureContext({
+    cert: privateKeyPemCert,
+  });
+  const secureSocket = new tls.TLSSocket(new net.Socket(), { secureContext });
```

Comment:
Don't forget to destroy this socket afterwards to prevent leaking sockets

Reply:
done